client/src/client.ts

```diff
@@ -19,6 +19,69 @@ export async function get_client(context: ExtensionContext): Promise<LanguageCli
                 workspace.createFileSystemWatcher('**/rebar.config'),
                 workspace.createFileSystemWatcher('**/rebar.lock')
             ]
+        },
+        middleware: {
+            executeCommand: async (command, args, next) => {
+                //Ask for user input if the argument contains {user_input: {type: string, text? : string}}
+                //Used by Wrangler
+                if (command.split(':').length >= 2
+                    && command.split(':')[1].startsWith("wrangler-")
+                    && args.length >= 1 
+                    && "user_input" in args[0] 
+                    && "type" in args[0].user_input) 
+                {
+                    var input: string;
+                    switch (args[0].user_input.type) {
+                        case "variable":
+                            input = await window.showInputBox({
+                                placeHolder: args[0].user_input.text ?? "New name", validateInput: (value) => {
+                                    if (!/^[A-Z][\_a-zA-Z0-9\@]*$/.test(value)) {
+                                        return "Name must be a valid Erlang variable name";
+                                    }
+                                    return null;
+                                }
+                            });
+                            break;
+                        case "atom":
+                            input = await window.showInputBox({
+                                placeHolder: args[0].user_input.text ?? "New name", validateInput: (value) => {
+                                    if (!(/^[a-z][\_a-zA-Z0-9\@]*$/.test(value) || /^[\'][\_a-zA-Z0-9\@]*[\']$/.test(value))) {
+                                        return "Name must be a valid Erlang atom";
+                                    }
+                                    return null;
+                                }
+                            });
+                            break;
+                        case "macro":
+                            input = await window.showInputBox({
+                                placeHolder: args[0].user_input.text ?? "New name", validateInput: (value) => {
```

Comment:
Note: if you use `value` instead of `placeHolder` it will let you edit the proposal.  The `placeHolder` is just a prompt, you either accept it or type from scratch